src/IotHub/IotHub/IotHub/Routing/AddAzureRmIotHubMessageEnrichment.cs

```diff
@@ -0,0 +1,123 @@
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, HelpMessage = "Endpoint(s) to apply enrichments to.")]
+        [Parameter(Mandatory = true, ParameterSetName = ResourceParameterSet, HelpMessage = "Endpoint(s) to apply enrichments to.")]
+        [ValidateNotNullOrEmpty]
+        public string[] Endpoint { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(this.Key, Properties.Resources.AddIotHubRoute))
+            {
+                IotHubDescription iotHubDescription;
+                if (ParameterSetName.Equals(InputObjectParameterSet))
+                {
+                    this.ResourceGroupName = this.InputObject.Resourcegroup;
+                    this.Name = this.InputObject.Name;
+                    iotHubDescription = IotHubUtils.ConvertObject<PSIotHub, IotHubDescription>(this.InputObject);
+                }
+                else
+                {
+                    if (ParameterSetName.Equals(ResourceIdParameterSet))
+                    {
+                        this.ResourceGroupName = IotHubUtils.GetResourceGroupName(this.ResourceId);
+                        this.Name = IotHubUtils.GetIotHubName(this.ResourceId);
+                    }
+
+                    iotHubDescription = this.IotHubClient.IotHubResource.Get(this.ResourceGroupName, this.Name);
+                }
+
+                if(iotHubDescription.Properties.Routing.Enrichments == null)
+                {
+                    iotHubDescription.Properties.Routing.Enrichments = new List<EnrichmentProperties>();
+                }
+
+                if(!iotHubDescription.Properties.Routing.Enrichments.Any(x => x.Key.Equals(this.Key.Trim(), StringComparison.OrdinalIgnoreCase)))
+                {
+                    IList<string> endpointNames = new List<string>();
+                    foreach (string endpoint in this.Endpoint)
+                    {
+                        endpointNames.Add(endpoint.Trim());
+                    }
+
+                    iotHubDescription.Properties.Routing.Enrichments.Add(
+                        new EnrichmentProperties(
+                            this.Key.Trim(),
+                            this.Value.Trim(),
+                            endpointNames
+                        ));
+
+                    this.IotHubClient.IotHubResource.CreateOrUpdate(this.ResourceGroupName, this.Name, iotHubDescription);
+                    IotHubDescription updatedIotHubDescription = this.IotHubClient.IotHubResource.Get(this.ResourceGroupName, this.Name);
+                    this.WriteObject(IotHubUtils.ToPSEnrichmentMetadata(updatedIotHubDescription.Properties.Routing.Enrichments.FirstOrDefault(x => x.Key.Equals(this.Key, StringComparison.OrdinalIgnoreCase))), false);
+                }
+                else
+                {
+                    throw new ArgumentException(string.Format("The enrichment's key \"{0}\" already exists.", this.Key));
```

Comment:
This should be a resource string (as with all customer-facing strings)

Reply:
Fixed